File: app/src/main/java/intellij/class13/deliverableexercise/hash/java/Student.java
package intellij.class13.deliverableexercise.hash.java;

import java.util.Objects;

public class Student {
    private String name;
    private String lastName;
    private Integer studentCode;

    public Student(String name, String lastName, Integer studentCode) {
        this.name = name;
        this.lastName = lastName;
        this.studentCode = studentCode;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Integer getStudentCode() {
        return studentCode;
    }

    public void setStudentCode(Integer studentCode) {
        this.studentCode = studentCode;
    }

    //Equals padrão que compara o objeto inteiro
//    @Override
//    public boolean equals(Object o) {
//        if (this == o) return true;
//        if (o == null || getClass() != o.getClass()) return false;
//        Student student = (Student) o;
//        return Objects.equals(nome, aluno.nome) &&
//                Objects.equals(numeroDoAluno, aluno.numeroDoAluno);
//    }

    //Equals que compara apenas o Código do Aluno do objeto
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Student student = (Student) o;
        return Objects.equals(studentCode, student.studentCode);
    }

    @Override
    public String toString() {
        return "'" + name + " " + lastName + '\'';
    }
}

File: src/class13/deliverableexercise/kotlin/Student.kt
package class13.deliverableexercise.kotlin

data class Student(
    var name: String,
    var lastName: String,
    var studentCode: Int
) {

    //Equals padrão que compara o objeto inteiro
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as Student
//        if (name != other.name) return false
//        if (lastName != other.lastName) return false
//        if (studentCode != other.studentCode) return false
//        return true
//    }

    //Equals que compara apenas o Código do Aluno do objeto
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val student = other as Student
        return studentCode == student.studentCode
    }

    override fun toString(): String {
        return "'$name $lastName'"
    }
}